src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractTableDataSource.cs

```diff
@@ -4,36 +4,47 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
+using Microsoft.CodeAnalysis;
 using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
 {
+    /// <summary>
+    /// Base implementation of ITableDataSource
+    /// </summary>
     internal abstract class AbstractTableDataSource<TData> : ITableDataSource
     {
-        protected readonly object Gate;
-        protected readonly Dictionary<object, AbstractTableEntriesFactory<TData>> Map;
+        private readonly object _gate;
+        private readonly Dictionary<object, TableEntriesFactory<TData>> _map;
+        private readonly Dictionary<object, object> _aggregateKeyMap;
```

Comment:
Can you please add a comment about what is stored in the key/value for this map?